jfxtras-agenda/src/main/java/jfxtras/internal/scene/control/skin/agenda/base24hour/AppointmentAbstractPane.java

```diff
@@ -333,18 +337,20 @@ else if ( (dragPickupInDayHeader && dragDropInDayBody) ) {
 			appointment.setWholeDay(false);
 
 			// if this is a task
+            boolean changed = false;
 			if (appointment.getStartLocalDateTime() != null && appointment.getEndLocalDateTime() == null) {
 				// set the drop time as the task time
 				appointment.setStartLocalDateTime(dragDropDateTime );
-				layoutHelp.callAppointmentChangedCallback(appointment);
+                changed = true;
 			}
 			else {
 				// simply add the duration, but without time
 				Period period = Period.between(dragPickupDateTime.toLocalDate(), dragDropDateTime.toLocalDate());
 				appointment.setStartLocalDateTime( appointment.getStartLocalDateTime().toLocalDate().plus(period).atStartOfDay() );
 				appointment.setEndLocalDateTime( appointment.getEndLocalDateTime().toLocalDate().plus(period).plusDays(1).atStartOfDay() );
-				layoutHelp.callAppointmentChangedCallback(appointment);
+                changed = true;
 			}
+            if (changed) layoutHelp.callAppointmentChangedCallback(appointment);
```

Comment:
OK.  I'll make sure I don't use single line if in your code in the future.  I'll even think about adopting that convention too...